src/pybind/mgr/dashboard/frontend/src/app/shared/components/deletion-button/deletion-button.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component, EventEmitter, Input, OnInit, Output, TemplateRef,
+  ViewChild
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { BsModalRef, BsModalService } from 'ngx-bootstrap';
+import { Observable } from 'rxjs/Observable';
+import { SubmitButtonComponent } from '../submit-button/submit-button.component';
+
+@Component({
+  selector: 'cd-deletion-button',
+  templateUrl: './deletion-button.component.html',
+  styleUrls: ['./deletion-button.component.scss']
+})
+export class DeletionButtonComponent implements OnInit {
+  @ViewChild(SubmitButtonComponent) submitButton: SubmitButtonComponent;
+  @Input() metaType: string;
+  @Input() pattern: string;
```

Comment:
Does it make sense to set a default value here?

Reply:
removed it